jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/HttpReceiver.java

```diff
@@ -440,7 +441,13 @@ protected boolean responseFailure(Throwable failure)
         // Mark atomically the response as completed, with respect
         // to concurrency between response success and response failure.
         if (exchange.responseComplete(failure))
-            return abort(exchange, failure);
+        {
+            // We can use a Promise.Completable and join it here only
+            // because the implementation of abort() is synchronous.
+            Promise.Completable<Boolean> promise = new Promise.Completable<>();
+            abort(exchange, failure, promise);
+            return promise.join();
```

Comment:
Don't like it. I think we can make `responseFailure()` take a `Promise` like everywhere else.

Reply:
Done.